cpp/src/sort/sort.cu

```diff
@@ -45,8 +45,7 @@ std::unique_ptr<table> sort_by_key(table_view const& values,
   CUDF_EXPECTS(values.num_rows() == keys.num_rows(),
                "Mismatch in number of rows for values and keys");
 
-  auto sorted_order = detail::sorted_order(
-    keys, column_order, null_precedence, stream, rmm::mr::get_current_device_resource());
+  auto sorted_order = detail::sorted_order(keys, column_order, null_precedence, stream, mr);
```

Comment:
This change seems wrong to me. The `sorted_order` is only a temporary variable here and so should not use the `mr` to create it.

Reply:
Thanks for clarifying. Corrected.